seirmo/figures.py

```diff
@@ -0,0 +1,99 @@
+#
+# This file is part of seirmo (https://github.com/SABS-R3-Epidemiology/seirmo/)
+# which is released under the BSD 3-clause license. See accompanying LICENSE.md
+# for copyright notice and full license details.
+#
+
+import plotly.graph_objects as go
+
+
+class IncidenceNumberPlot(object):
+    """
+
+    Creates incidence number plots for the app
+
+    """
+    def __init__(self):
+        super(IncidenceNumberPlot, self).__init__()
+
+        # Instantiate an empty plotly figure
+        self._fig = go.Figure()
+
+        """
+        Add a bar plot trace for the given incidence number data
+        in the figure
+
+        Parameters
+        ----------
+        data: pandas.DataFrame
+        time_key: docstring
+        inc_key: docstring
+        """
+    def add_data(self, data, time_key='Time',
+                 inc_key='Incidence Number'):
+
+        # Plot a bar chart for the data
+        self._fig.add_trace(
+            go.Bar(
+                x=data[time_key],
+                y=data[inc_key],
+                name='cases'
+            )
+        )
+
+        # Add axis labels; Raise error when the x and y axis labels
+        # do not match the old ones if they already exist;
+        # otherwise update the labels
+        old_x_label = self._fig['layout']['xaxis']['anchor']
```

Comment:
Can you put this axis label checking into a separate (private) method
`_update_axis_labels(x_label, y_label)`?

this method does exactly what you do from line 44 to 57, but then you can just call the method in both `add_data` and `add_simulation`